### PR TITLE
[cloud-provider-*] Minimize RBAC permissions in the ClusterRole

### LINKED ISSUE #8517: Minimize RBAC permissions of platform components
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Kubernetes CIS [we recommend not using wildcard rights in Role/ClusterRole](https://github.com/aquasecurity/kube-bench/blob/0f8dfaf1156652b556c913aa8e78fccbb1dfd186/cfg/rke-cis-1.24/policies.yaml#L29-L35). 
However, the DKP platform does not take this recommendation into account everywhere. 
For example:
- dex has [verbs: *](https://github.com/deckhouse/deckhouse/blob/main/modules/150-user-authn/templates/dex/rbac-for-us.yaml#L53)
- upmeter-agent [has wildcard rights to all resources](https://github.com/deckhouse/deckhouse/blob/main/modules/500-upmeter/templates/upmeter-agent/rbac-for-us.yaml#L63-L73) in d8-upmeter namespace

Therefore, the DKP does not fully comply with the Kubernetes CIS recommendations.

### Proposed Solution

Grant specific required permissions to objects in the Kubernetes API.

### Additional Information

You might be interested in the [audit2rbac](https://github.com/liggitt/audit2rbac) tool to minimize permissions for the modules.

### PR DESCRIPTION
**Pull request by artbin**
## Description

Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.

## Why do we need it, and what problem does it solve?

Fixes #8517

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-aws
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-vsphere
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-azure
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-gcp
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-vcd
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-yandex
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-openstack
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

```changes
section: cloud-provider-zvirt
type: fix
summary: Minimize RBAC permissions by removing the wildcard("*") from ClusterRole rules.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/static'...

**Comment by artbin** [comment]
/e2e/abort vclouddirector;Standard;containerd;1.27 9874538858 9874538858-1-con-1-27 /sys/deckhouse-oss/install:pr8969

**Comment by artbin** [comment]
/e2e/abort static;Static;containerd;1.27 9874539726 9874539726-1-con-1-27 /sys/deckhouse-oss/install:pr8969 astra@192.168.199.122

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/aws'...

**Comment by artbin** [comment]
/e2e/abort vclouddirector;Standard;containerd;1.27 9888595355 9888595355-2-con-1-27 /sys/deckhouse-oss/install:pr8969 ubuntu@192.168.254.95

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by artbin** [comment]
/e2e/abort openstack;Standard;containerd;1.27 9950499571 9950499571-1-con-1-27 /sys/deckhouse-oss/install:pr8969 redos@45.12.229.213

**Comment by artbin** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 9950499577 9950499577-1-con-1-27 /sys/deckhouse-oss/install:pr8969 ec2-user@3.71.212.2

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/aws'...

**Comment by artbin** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 9952223778 9952223778-1-con-1-27 /sys/deckhouse-oss/install:pr8969 ec2-user@18.194.141.49

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by artbin** [comment]
/e2e/abort openstack;Standard;containerd;1.27 10059390262 10059390262-1-con-1-27 /sys/deckhouse-oss/install:pr8969

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by z9r5** [comment]
/e2e/abort openstack;Standard;containerd;1.27 10060416502 10060416502-1-con-1-27 /sys/deckhouse-oss/install:pr8969

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Review by aleksey-su — APPROVED** [review]

File: ee/modules/030-cloud-provider-openstack/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 1]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline ee/modules/030-cloud-provider-openstack/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add deletecollection too.
 
https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: ee/modules/030-cloud-provider-vcd/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 2]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline ee/modules/030-cloud-provider-vcd/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: ee/modules/030-cloud-provider-vsphere/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 3]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline ee/modules/030-cloud-provider-vsphere/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: ee/modules/030-cloud-provider-zvirt/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 4]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline ee/modules/030-cloud-provider-zvirt/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: modules/030-cloud-provider-aws/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 5]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline modules/030-cloud-provider-aws/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: modules/030-cloud-provider-azure/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 6]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline modules/030-cloud-provider-azure/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: modules/030-cloud-provider-gcp/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 7]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline modules/030-cloud-provider-gcp/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: modules/030-cloud-provider-yandex/templates/cloud-controller-manager/rbac-for-us.yaml [hunk 8]
```diff
@@ -25,7 +25,13 @@ rules:
   resources:
   - nodes
   verbs:
-  - '*'
+  - create
+  - delete
```
Comment (name212) [inline modules/030-cloud-provider-yandex/templates/cloud-controller-manager/rbac-for-us.yaml:29]: please add `deletecollection` too.

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

**Review by name212 — CHANGES_REQUESTED** [review]
please add `deletecollection` in the all cloud providers for `node` object

https://kubernetes.io/docs/reference/access-authn-authz/authorization/#request-verb-non-resource

File: ee/modules/030-cloud-provider-openstack/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 1 above)
Comment (artbin) [reply ee/modules/030-cloud-provider-openstack/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: ee/modules/030-cloud-provider-vcd/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 2 above)
Comment (artbin) [reply ee/modules/030-cloud-provider-vcd/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: ee/modules/030-cloud-provider-vsphere/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 3 above)
Comment (artbin) [reply ee/modules/030-cloud-provider-vsphere/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: ee/modules/030-cloud-provider-zvirt/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 4 above)
Comment (artbin) [reply ee/modules/030-cloud-provider-zvirt/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: modules/030-cloud-provider-aws/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 5 above)
Comment (artbin) [reply modules/030-cloud-provider-aws/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: modules/030-cloud-provider-azure/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 6 above)
Comment (artbin) [reply modules/030-cloud-provider-azure/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

File: modules/030-cloud-provider-gcp/templates/cloud-controller-manager/rbac-for-us.yaml (hunk 7 above)
Comment (artbin) [reply modules/030-cloud-provider-gcp/templates/cloud-controller-manager/rbac-for-us.yaml:29]: added

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/gcp'...

**Comment by artbin** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 10349799330 10349799330-1-con-1-27 /sys/deckhouse-oss/install:pr8969 ec2-user@3.72.189.107

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by artbin** [comment]
/e2e/abort vsphere;Standard;containerd;1.27 10351569840 10351569840-1-con-1-27 /sys/deckhouse-oss/install:pr8969 ubuntu@10.220.203.26

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Review by name212 — APPROVED** [review]

**Pull request merged**